discretisedfield/tests/test_tools.py

```diff
@@ -64,56 +64,96 @@ def test_topological_charge(method):
             getattr(dft, function)(f.sel("z"), method="wrong")
 
 
-def test_emergent_magnetic_field():
-    p1 = (0, 0, 0)
-    p2 = (10, 10, 10)
-    cell = (2, 2, 2)
+@pytest.mark.parametrize("ndim", [1, 2, 3, 4])
+@pytest.mark.parametrize("nvdim", [1, 2, 3, 4])
+def test_emergent_magnetic_field(ndim, nvdim):
+    p1 = (0,) * ndim
+    p2 = (10,) * ndim
+    cell = (2,) * ndim
+    value = (0,) * nvdim
```

Comment:
If you are initialising with zeros then I don't think this is needed as the default value is zero anyway. As long as we have `nvdim=nvdim` when initalising.

Reply:
Nicely spotted! 😃